commit dc202e4f589834e3bef6236ae5657c244811dd94 (refs/changes/96/2296/2)
Author: Julien Gilli <julien.gilli@joyent.com>
Date:   2017-07-31T16:24:39-07:00 (2 years, 2 months ago)
    
    joyent/node-triton#217 `triton volume ls -l` should output a `RESOURCE` column

diff --git a/CHANGES.md b/CHANGES.md
index 2fba584..50044cc 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -7,6 +7,8 @@ Known issues:
 
 ## not yet released
 
+- [joyent/node-triton#217] `triton volume ls -l` should output a `RESOURCE`
+  column.
 - [joyent/node-triton#173], [joyent/node-triton#174] and
   [joyent/node-triton#175] Add support for creating and managing NFS shared
   volumes. New `triton volume` commands are available:
diff --git a/lib/do_volume/do_list.js b/lib/do_volume/do_list.js
index 9b16a82..147650f 100644
--- a/lib/do_volume/do_list.js
+++ b/lib/do_volume/do_list.js
@@ -31,7 +31,7 @@ var validFilters = [
 var columnsDefault = 'shortid,name,size,type,state,age';
 
 // columns default with -l
-var columnsDefaultLong = 'id,name,size,type,state,created';
+var columnsDefaultLong = 'id,name,size,type,resource,state,created';
 
 // sort default with -s
 var sortDefault = 'created';
@@ -98,6 +98,10 @@ function do_list(subcmd, opts, args, callback) {
 
                         created = new Date(volume.create_timestamp);
 
+                        if (volume.filesystem_path !== undefined) {
+                            volume.resource = volume.filesystem_path;
+                        }
+
                         volume.shortid = volume.id.split('-', 1)[0];
                         volume.created = volume.create_timestamp;
                         volume.age = common.longAgo(created, now);
@@ -142,6 +146,8 @@ do_list.help = [
     'for convenience):',
     '    shortid*           A short ID prefix.',
     '    age*               Approximate time since created, e.g. 1y, 2w.',
+    '    resource*          A locator usable by clients to make use of the ',
+    '                       volume\'s resources',
     ''
     /* END JSSTYLED */
 ].join('\n');
